blog/2025-12-21-openhab-5-1-release.md

```diff
@@ -0,0 +1,95 @@
+date: "2025-12-21T20:00:00+01:00"
+previewimage: /uploads/2025-12-21-openhab-5-1-release/header.webp
+tags:
+  - news
+excerpt: >-
+  
+---
+
+# Activity
+
+# Highlights
+
+As you might have already noticed in the activity statistics, there was much development on openHAB - too much to list everything.
+Please refer to the [**release notes**](https://github.com/openhab/openhab-distro/releases/tag/5.1.0) to check out what's new and noteworthy (e.g. breaking changes).
+
+In the following sections, our maintainers and contributors are introducing some new features you don't want to miss!
+
+## Event Source Tracking
+
+@ccutrer
+
+## Core Runtime Enhancements
+
+## Main UI Enhancements
+
+_Florian Hotze ([@florian-h05](https://github.com/florian-h05)), openHAB Maintainer_
+
+The openHAB 5.1 release is a very special release for Main UI.
+There may not be as many new features and visual changes as usual, but we've been very busy working "under the hood", future-proofing our codebase.
+
+### Vue 3 Upgrade
+
+Main UI has been introduced five years ago with the openHAB 3.0 relase.
+Especially in the fast-paced world of web development, a codebase can quickly get stuck in a "legacy" state in that time.
+We therefore are even more thrilled to announce that we have completed a massive undertaking:
+
+Upgrading from Vue 2 to Vue 3, Framework7 5 to Framework7 7, Vuex to Pinia, and introducing TypeScript into Main UI's codebase.
+
+The first steps on that long path were taken in May this year, resulting in the submission of a gigantic PR at the begin of September, touching about 10,000 lines of code.
+After two months of testing, fixing bugs and reviewing the changes, that PR was merged end of October.
+Since then, we've commited another 150 changes fixing the remaining bugs, refactoring our codebase to TypeScript and implementing new features.
+
+Thousand thanks to Jeff James ([@jsjames](https://github.com/jsjames)) for initially starting that effort, and the great team work in the last months - and welcome to the Main UI maintainer team!
+
+### Thing & Item Code Tabs
+
+@jimtng: https://github.com/openhab/openhab-webui/pull/3180
+
+### oh-video Widget
+
+@digitaldan: https://github.com/openhab/openhab-webui/pull/3368
+
+### Visualize semantic model tree for Items
+
+@mherwerge: https://github.com/openhab/openhab-webui/pull/3227
+
+### Developer Sidebar Search
+
+@jimtng: https://github.com/openhab/openhab-webui/pull/3289
+
+### Things, Items & Rules Filtering
+
+@jimtng: https://github.com/openhab/openhab-webui/pull/3308
+
+## Blockly Enhancements
+
+Blockly has also received a major upgrade, including several plugins updates, which fixes a number of issues with our new Vue 3 Main UI.
+Additionally, a new logic block allows to link multiple AND or OR statements comfortably without the clutter of deep nesting.
+
+## Basic UI Enhancements
+
+## Add-on Enhancements
+
+openHAB 5.1 brings xx new add-ons, adding support for new devices:
+
```

Comment:
- [HomeKit Binding](/addons/bindings/homekit/) This is a new binding that allows pairing with HomeKit accessories and **imports** their services as channel groups and their respective service- characteristics as channels.
Do not confuse this with the [HomeKit System Integration](/addons/integrations/homekit/) which **exports** openHAB Items to a HomeKit controller.